packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -50,3 +76,132 @@ void writeFlutterPluginsList() {
     }
   }
 }
+
+const String _androidPluginRegistryTemplate = '''package io.flutter.plugins;
+
+import io.flutter.app.FlutterActivity;
+
+{{#plugins}}
+import {{package}}.{{class}};
+{{/plugins}}
+
+/**
+ * This file is generated. Do not edit.
+ */
+
+public class PluginRegistry {
+{{#plugins}}
+    {{class}} {{name}};
+{{/plugins}}
+
+    public void registerAll(FlutterActivity activity) {
+{{#plugins}}
+        {{name}} = {{class}}.register(activity);
+{{/plugins}}
+    }
+}
+''';
+
+void _writeAndroidPluginRegistry(List<Plugin> plugins) {
+  final List<Map<String, dynamic>> androidPlugins = plugins
+      .where((Plugin p) => p.androidPackage != null && p.pluginClass != null)
+      .map((Plugin p) => <String, dynamic>{
+          'name': p.name,
+          'package': p.androidPackage,
+          'class': p.pluginClass,
+      });
+  final Map<String, dynamic> context = <String, dynamic>{
+    'plugins': androidPlugins,
+  };
+
+  final String pluginRegistry =
+      new mustache.Template(_androidPluginRegistryTemplate).renderString(context);
+  final Directory registryDirectory =
+      fs.directory(fs.path.join('android', 'app', 'src', 'main', 'java', 'io', 'flutter', 'plugins'));
+  registryDirectory.createSync(recursive: true);
+  final File registryFile = registryDirectory.childFile('PluginRegistry.java');
+  registryFile.writeAsStringSync(pluginRegistry);
+}
+
+const String _iosPluginRegistryHeaderTemplate = '''//
+//  Generated file. Do not edit.
+//
+
+#ifndef PluginRegistry_h
+#define PluginRegistry_h
+
+#import <Flutter/Flutter.h>
+
+{{#plugins}}
+#import "{{class}}.h"
+{{/plugins}}
+
+@interface PluginRegistry : NSObject
+
+{{#plugins}}
+@property (readonly, nonatomic) {{class}}* {{name}};
+{{/plugins}}
+
+- (id) initWithController: (FlutterViewController*)controller;
+
+@end
+
+#endif /* PluginRegistry_h */
+''';
+
+const String _iosPluginRegistryImplementationTemplate = '''//
+//  Generated file. Do not edit.
+//
+
+#import "PluginRegistry.h"
+
+@implementation PluginRegistry
+
+{{#plugins}}
+@synthesize {{name}};
+{{/plugins}}
+
+- (id) initWithController:(FlutterViewController *)controller {
+  if (self = [super init]) {
+{{#plugins}}
+    {{name}} = [[{{class}} alloc] initWithController:controller];
```

Comment:
I'm kind of surprised this work, since you're mutating a readonly property here. I would put an underscore before {{name}} to make it clear you're not calling the synthesized accessor method, but rather mutating the synthesized member variable.